src/background/logger/InBrowserLogger.ts

```diff
@@ -0,0 +1,19 @@
+import { Logger } from "../../shared/logger/Logger";
+import { LogEntry } from "../../shared/types/logger/LogEntry";
+import { InBrowserPersistenceManager } from "./InBrowserPersistenceManager";
+
+export class InBrowserLogger extends Logger {
+  constructor(persistence: InBrowserPersistenceManager) {
+    super(persistence);
+  }
+
+  public async getAll(): Promise<LogEntry[] | null> {
+    const castPersistence = <InBrowserPersistenceManager>this.persistence;
```

Comment:
Why is it necessary to cast it to its original type? Maybe it's because you didn't specify an access modifier (public/private) in the constructor?

Reply:
Done